app/validators/program_validator.rb

```diff
@@ -1,17 +1,36 @@
 # frozen_string_literal: true
 
 class ProgramValidator < BaseValidator
```

Comment:
Trying to figure out the best change to make here.  `ProgramValidator` doesn't need to be a subclass of `ActiveModel::Validator` anymore
`BaseValidator#facility_code_in_weam`is only used by `EduProgramValidator`, so it can be moved there.
I'm thinking:
* make `EduProgramValidator`'s parent class `ActiveModel::Validator`
* move `#facility_code_in_weam` to `EduProgramValidator`
* remove `ProgramValidator` parent class declaration
* change `BaseValidator` to `SharedMessages` with `self.missing_facility_error_msg` to be called by `EduProgramValidator` and `ProgramValidator`.  This part I'm least confident in, but can't think of a much cleaner way to organize it.
Thoughts?

Reply:
changed